js/ui/windowManager.js

```diff
@@ -158,6 +159,17 @@ var EOSShellWobbly = new Lang.Class({
         binder('wobbly-object-movement-range', 'object-movement-range');
     },
 
+    _removeFromActorOnDisabled: function(actor) {
+        if (!this.enabled) {
+            if (this._wobblyEffectDisabledId !== -1) {
+                this.disconnect(this._wobblyEffectDisabledId);
+                this._wobblyEffectDisabledId = -1;
+            }
+
+            actor.remove_effect(this);
```

Comment:
This strikes me as an odd way of achieving the goal of the commit.
What ends up actually disabling the effect? The effect is created and added in `windowManager.js` before calling into `grabbedByMouse()`. If the idea is to remove it when it's ungrabbed, why not just do it from there as well?

Reply:
So, ClutterActorMeta has an 'enabled' property which for a ClutterEffect, will be flipped to FALSE when there is no more animation to do.

We cant remove it immediately on ungrab - there might still be force/velocity/deformation remaining in the model that has yet to play out.

Reply:
Presumably though, the `enabled` property is designed to achieve the same that you're trying to achieve here, no? Meaning to avoid doing any work when the animation is not running.
Would it be possible to adjust the effect so that it turns into a no-op when it's disabled, instead of forcibly removing it like this? It looks like that's the usual pattern for clutter effects.

Reply:
I could certainly rework the effect to just do nothing when `enabled` is not set.

The current approach is derived from what is typically done in compiz, which is that hooks on paint operations are removed if they are inactive to save CPU cycles on function call overhead. But its probably not a big deal in this case.